### PR TITLE
OPAL-2865 - add vcf store plugins system resources and a way to edit …

### PR DESCRIPTION
**Pull request by meek0**
…a project's vcf store service

### CODE REVIEW

File: opal-core-ws/src/main/java/org/obiba/opal/web/system/SystemResource.java [hunk 1]
```diff
@@ -235,4 +244,12 @@ public KeyStoreResource getKeyStoreResource() {
     resource.setKeyStore(systemKeyStoreService.getKeyStore());
     return resource;
   }
+
+  @GET
+  @Path("/plugins/vcf-stores")
+  public Response getVcfStores() {
```
Comment (ymarcon) [inline opal-core-ws/src/main/java/org/obiba/opal/web/system/SystemResource.java]: Not the right place to have this entry point. There is already a PluginsResource

File: opal-core-ws/src/main/java/org/obiba/opal/web/system/plugins/VCFStorePluginResource.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+  public enum PLUGIN_STATUS { RUNNING, STOPPED }
+
+  @Autowired
+  private OpalRuntime opalRuntime;
+
+  @GET
+  public Response get(@PathParam("name") String name) {
+    Optional<VCFStoreService> vcfService = getVCFService(name);
+    Response.ResponseBuilder response = Response.status(vcfService.isPresent() ? Response.Status.OK : Response.Status.NOT_FOUND);
```
Comment (ymarcon) [inline opal-core-ws/src/main/java/org/obiba/opal/web/system/plugins/VCFStorePluginResource.java]: This code is useless: make sure there is a mapper for the NoSuchElementException

File: opal-core-ws/src/main/java/org/obiba/opal/web/system/plugins/VCFStorePluginResource.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    vcfService.ifPresent(vcfStoreService -> response.entity(vcfStoreService.isRunning() ? PLUGIN_STATUS.RUNNING : PLUGIN_STATUS.STOPPED));
+    return response.build();
+  }
+
+  @PUT
+  public Response start(@PathParam("name") String name) {
+    Optional<VCFStoreService> vcfService = getVCFService(name);
+    Response.ResponseBuilder response = Response.status(vcfService.isPresent() ? Response.Status.OK : Response.Status.NOT_FOUND);
```
Comment (ymarcon) [inline opal-core-ws/src/main/java/org/obiba/opal/web/system/plugins/VCFStorePluginResource.java]: Same comment: optional etc is useless, let the exception go

File: opal-core-ws/src/main/java/org/obiba/opal/web/system/plugins/VCFStorePluginResource.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    vcfService.ifPresent(ServicePlugin::start);
+    return response.build();
+  }
+
+  @DELETE
+  public Response stop(@PathParam("name") String name) {
+    Optional<VCFStoreService> vcfService = getVCFService(name);
+    Response.ResponseBuilder response = Response.status(vcfService.isPresent() ? Response.Status.OK : Response.Status.NOT_FOUND);
```
Comment (ymarcon) [inline opal-core-ws/src/main/java/org/obiba/opal/web/system/plugins/VCFStorePluginResource.java]: Usually a delete is silent: trap any NoSuch... and return Response.noContent()

File: opal-core-ws/src/main/java/org/obiba/opal/web/system/plugins/VCFStorePluginResource.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import javax.ws.rs.*;
+import javax.ws.rs.core.Response;
+import java.util.Optional;
+
+@Component
+@Path("/system/plugins/vcf-store/{name}")
+public class VCFStorePluginResource {
```
Comment (ymarcon) [inline opal-core-ws/src/main/java/org/obiba/opal/web/system/plugins/VCFStorePluginResource.java]: there is nothing specific to the VCF store in this code. Use the already existing PluginResource instead.

**Comment by meek0** [comment]
Changes are done. Also create PR https://github.com/obiba/jennite/pull/2 to help the plugins resource filter the type vcf-store.

File: opal-core-ws/src/main/java/org/obiba/opal/web/system/SystemResource.java [hunk 6]
```diff
@@ -71,6 +62,9 @@
   @Autowired
   private ApplicationContext applicationContext;
 
+  @Autowired
+  private OpalRuntime opalRuntime;
```
Comment (ymarcon) [inline opal-core-ws/src/main/java/org/obiba/opal/web/system/SystemResource.java]: If it's not used, remove it.

File: opal-gwt-client/src/main/java/org/obiba/opal/web/gwt/app/GwtApp.gwt.xml [hunk 7]
```diff
@@ -19,7 +19,7 @@
 
   <!-- Logging -->
   <inherits name="com.google.gwt.logging.Logging"/>
-  <set-property name="gwt.logging.enabled" value="FALSE"/>
+  <set-property name="gwt.logging.enabled" value="TRUE"/>
```
Comment (ymarcon) [inline opal-gwt-client/src/main/java/org/obiba/opal/web/gwt/app/GwtApp.gwt.xml]: Do not commit this.

File: opal-gwt-client/src/main/java/org/obiba/opal/web/gwt/app/GwtApp.gwt.xml [hunk 8]
```diff
@@ -133,12 +133,12 @@
     </any>
   </replace-with>
 
-  <set-property name='user.agent' value='gecko1_8,safari,ie9,ie10'/>
+  <!--<set-property name='user.agent' value='gecko1_8,safari,ie9,ie10'/>-->
```
Comment (ymarcon) [inline opal-gwt-client/src/main/java/org/obiba/opal/web/gwt/app/GwtApp.gwt.xml]: Do not commit this.

File: opal-gwt-client/src/main/java/org/obiba/opal/web/gwt/app/GwtApp.gwt.xml [hunk 9]
```diff
@@ -133,12 +133,12 @@
...
 
   <!-- This will limit the compile to a single browser: -->
   <!-- Firefox 3.x -->
   <!--<set-property name='user.agent' value='gecko1_8'/>-->
   <!-- Chrome -->
-  <!--<set-property name='user.agent' value='safari'/>-->
+  <set-property name='user.agent' value='safari'/>
```
Comment (ymarcon) [inline opal-gwt-client/src/main/java/org/obiba/opal/web/gwt/app/GwtApp.gwt.xml]: Do not commit this.

File: opal-ws/src/main/java/org/obiba/opal/web/services/PluginResource.java [hunk 10]
```diff
@@ -31,11 +29,21 @@
...
+  @Autowired
   private OpalRuntime opalRuntime;
 
   @GET
   public Plugins.PluginDto get() {
     return Dtos.asDto(opalRuntime.getPlugin(name));
   }
 
+  @Path("/{service}")
+  public VCFStorePluginResource vcfStoreService(@PathParam("service") String service) {
```
Comment (ymarcon) [inline opal-ws/src/main/java/org/obiba/opal/web/services/PluginResource.java]: A plugin is not necessarily about VCF.

Comment (ymarcon) [inline opal-ws/src/main/java/org/obiba/opal/web/services/PluginResource.java]: See ServicePlugin interface.

File: opal-ws/src/main/java/org/obiba/opal/web/services/VCFStorePluginResource.java [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+import org.springframework.stereotype.Component;
+
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.PUT;
+import javax.ws.rs.core.Response;
+
+@Component
+@Scope(ConfigurableBeanFactory.SCOPE_PROTOTYPE)
+public class VCFStorePluginResource {
```
Comment (ymarcon) [inline opal-ws/src/main/java/org/obiba/opal/web/services/VCFStorePluginResource.java]: Make a resource over ServicePlugin interface instead.

**Comment by meek0** [comment]
Made some changes and also created PR https://github.com/obiba/jennite/pull/3

**Pull request merged**